src/test/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParserTest.java

```diff
@@ -91,374 +46,310 @@ public static void setUp() {
-        SuggestionItem cs2103Week3 = new SuggestionItemImpl(toCs2103Week3.getStringRepresentation(), null);
-
-        List<SuggestionItem> expectedSuggestions = new ArrayList<>();
-        expectedSuggestions.add(cs2103);
-        expectedSuggestions.add(cs2103Week1);
-        expectedSuggestions.add(cs2103Week2);
-        expectedSuggestions.add(cs2103Week3);
-        expectedSuggestions.add(cs2103Week1Lecture);
+        // Expected suggestions
+        List<SuggestionItem> expectedSuggestions = SuggestionTestUtil.getExpectedSugForCs2103tPathInput();
 
+        // Actual suggestions
         List<SuggestionItem> suggestions = model.getSuggestions();
 
-        // check display text
-        for (int i = 0; i < expectedSuggestions.size(); i++) {
-            SuggestionItem suggestion = suggestions.get(i);
-            SuggestionItem expectedSuggestion = expectedSuggestions.get(i);
-            assertEquals(expectedSuggestion.getProperty("displayText"), suggestion.getProperty("displayText"));
-        }
-
-        List<String> expectedInputs = new ArrayList<>();
-        expectedInputs.add(COMMAND_WORD + " " + toCs2103.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + toCs2103Week1.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + toCs2103Week2.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + toCs2103Week3.getStringRepresentation());
-        expectedInputs.add(COMMAND_WORD + " " + toCs2103Week1Lecture.getStringRepresentation());
-
-        for (int i = 0; i < expectedInputs.size(); i++) {
-            SuggestionItem suggestionItem = suggestions.get(i);
-            String expectedInput = expectedInputs.get(i);
-            suggestionItem.getAction().run();
-            String input = model.getInput();
-            assertEquals(expectedInput, input);
-        }
+        // Test suggestions
+        SuggestionTestUtil.testSuggestions(expectedSuggestions, suggestions);
+
+        // Expected inputs
+        List<String> expectedInputs = SuggestionTestUtil.getExpectedInputsForCs2103tPathInput(userInputWithoutPath);
+
+        // Test inputs
+        SuggestionTestUtil.testInputs(expectedInputs, suggestions, model);
     }
 
     @Test
-    public void parse_correctedAbsolutePathWithPrefix_returnsDeleteSuggestionCommand() {
-        String arg = " " + PREFIX_TITLE + " /CS2104";
-        model.setInput(COMMAND_WORD + arg);
-        Optional<? extends SuggestionCommand> commandCorrectedPath = deleteSuggestionCommandParser.parse(arg);
-        assertTrue(commandCorrectedPath.get() instanceof DeleteSuggestionCommand);
+    public void parse_correctedCmdcorrectAbsolutePathWithoutPrefix_returnsDeleteSuggestionCommand() {
+        String userInputWithoutPath = "dele ";
+        String userInput = userInputWithoutPath + toCs2103t.getStringRepresentation();
+        String arg = userInput.replace("dele", "");
```

Comment:
Is this necessary? Why not just hardcode the `arg` string? It'll make your intention more explicit imo.